Sources_v3/Workers/Background/MissingEventsReplayer.swift

```diff
@@ -0,0 +1,100 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+import Foundation
+
+/// The type is designed to obtain missing events that happened in watched channels while user
+/// was not connected to the web-socket.
+///
+/// The object listenes for `ConnectionStatusUpdated` events
+/// and remembers the `CurrentUserDTO.lastReceivedEventDate` when status becomes `connecting`.
+///
+/// When the status becomes `connected` the `/sync` endpoint is called
+/// with `lastReceivedEventDate` and `cids` of watched channels.
+///
+/// We remember `lastReceivedEventDate` when state becomes `connecting` to catch the last event date
+/// before the `HealthCheck` override the `lastReceivedEventDate` with the recent date.
+///
+class MissingEventsReplayer<ExtraData: ExtraDataTypes>: Worker {
```

Comment:
Do you have some alternative naming ideas?? :D

Reply:
Hm.. 🤔  How about `MissingEventsPublisher`?

Reply:
It's a little bit confusing with relation to `Combine` but I think it's better than `Replayer` :)